### PR TITLE
Add com.cburch.Logisim

### PR DESCRIPTION
**Pull request by Dimitry-bit**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I have [built][build] and tested the submission locally.
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** ( Project is under GNU V2 [Copyright and authorship](http://www.cburch.com/logisim/download.html) )
- [ ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flatpak.org/en/latest/first-build.html

### CODE REVIEW

File: com.cburch.Logisim.metainfo.xml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+      build larger circuits from smaller subcircuits, and to draw bundles of wires with a single
+      mouse drag, Logisim can be used (and is used) to design and simulate entire CPUs for
+      educational purposes.</p>
+    <p>Logisim is used by students at colleges and universities around the world in many types of
+      classes, ranging from a brief unit on logic in general-education computer science surveys, to
+      computer organization courses, to full-semester courses on computer architecture.</p>
+  </description>
+
+  <screenshots>
+    <screenshot type="default">https://i.imgur.com/Vgm01Ip.png</screenshot>
```
Comment (hfiguiere) [inline com.cburch.Logisim.metainfo.xml:29]: imgur is not a reliable storage for screenshots.

File: com.cburch.Logisim.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+sdk: org.freedesktop.Sdk
+command: logisim
+sdk-extensions:
+  - org.freedesktop.Sdk.Extension.openjdk17
+finish-args:
+  # X11 + XShm access
+  - --share=ipc
+  - --socket=fallback-x11
+  # Wayland access
+  - --socket=wayland
```
Comment (hfiguiere) [inline com.cburch.Logisim.yml:13]: does it really support wayland?

File: com.cburch.Logisim.yml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  # GPU acceleration if needed
+  - --device=dri
+  # Needs to save files locally
+  - --filesystem=host
```
Comment (hfiguiere) [inline com.cburch.Logisim.yml:17]: that remaind a very broad filesystem permission

**Comment by hfiguiere** [comment]
bot, build com.cburch.Logisim

File: com.cburch.Logisim.metainfo.xml (hunk 1 above)
Comment (Dimitry-bit) [reply com.cburch.Logisim.metainfo.xml]: Hosted screenshots on GitHub.

File: com.cburch.Logisim.yml (hunk 2 above)
Comment (Dimitry-bit) [reply com.cburch.Logisim.yml]: I misunderstood `--socket=wayland` purpose, my bad.

File: com.cburch.Logisim.yml (hunk 3 above)
Comment (Dimitry-bit) [reply com.cburch.Logisim.yml]: Revoked it to include home directory only.

**Review by bbhtt — COMMENTED** [review]
Desktop file, icon and appdata should be upstream.

**Comment by bbhtt** [comment]
Crashes immediately:

```
╭─bbhtt@fedora in ~ took 13s
→ flatpak run com.cburch.Logisim                                                              
Nov 21, 2023 6:24:54 AM java.util.prefs.FileSystemPreferences$1 run
INFO: Created user preferences directory.
Exception in thread "main" java.awt.HeadlessException: 
No X11 DISPLAY variable was set,
but this program performed an operation which requires it.
	at java.desktop/sun.awt.HeadlessToolkit.getMenuShortcutKeyMask(HeadlessToolkit.java:135)
	at com.cburch.logisim.gui.menu.MenuFile.<init>(MenuFile.java:38)
	at com.cburch.logisim.gui.menu.LogisimMenuBar.<init>(LogisimMenuBar.java:87)
	at com.cburch.logisim.gui.start.Startup.run(Startup.java:159)
	at com.cburch.logisim.Main.main(Main.java:36)

```

**Comment by Dimitry-bit** [comment]
I tested it on X11 and on wayland through Xwayland , and it ran fine. Keep in mind that Logisim supports X11 only, so Xorg or Xwayland should be install for it to function properly.

**Comment by bbhtt** [comment]
> I tested it on X11 and on wayland through Xwayland , and it ran fine. Keep in mind that Logisim supports X11 only, so Xorg or Xwayland should be install for it to function properly.

GNOME X11 is working fine, with GNOME Wayland session I see the crash.

**Comment by bbhtt** [comment]
Ah, figured the issue, I was testing the old build https://github.com/flathub/flathub/pull/4716#issuecomment-1819271377. There hasn't been a new build with `socket=wayland` removed. Having that triggers the crash.

**Comment by bbhtt** [comment]
bot, build com.cburch.Logisim

**Comment by bbhtt** [comment]
> Build [82895 successful](https://buildbot.flathub.org/#/builders/6/builds/82895) To test this build, install it from the testing repository:
> 
> ```
> flatpak install --user https://dl.flathub.org/build-repo/65529/com.cburch.Logisim.flatpakref
> ```

Works now

**Pull request closed without merging**